### PR TITLE
Bump tendermint-rs dependencies to 0.23.6

### PR DESCRIPTION
**Pull request by romac**
Closes: #2045

## Description

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules). 
- [x] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: relayer/src/chain/cosmos.rs [hunk 1]
```diff
@@ -304,7 +304,7 @@ impl CosmosSdkChain {
         crate::telemetry!(query, self.id(), "query_consensus_params");
 
         Ok(self
-            .block_on(self.rpc_client.genesis())
+            .block_on(self.rpc_client.genesis::<serde_json::Value>())
```
Comment (hu55a1n1) [inline relayer/src/chain/cosmos.rs:307]: Doesn't look like this `query_consensus_params()` function is used anywhere. Should we remove it?

**Review by hu55a1n1 — APPROVED** [review]
Approved with a minor suggestion. 🙏

File: relayer/src/chain/cosmos.rs (hunk 1 above)
Comment (romac) [reply relayer/src/chain/cosmos.rs:307]: Good catch!

Comment (romac) [reply relayer/src/chain/cosmos.rs]: efd1950b1c10bf2d264e02203a95e2ebdff26cbe

**Pull request merged**